### PR TITLE
[IMP] account : cash discount: set up default accounts and configs

### PR DESCRIPTION
**Pull request by casp-odoo**
The early payment cash discount functionality was merged in 16.0.

This PR allows for the behavior to be as localization specific as possible. 

This concerns :
- The tax computation (some countries leave it untouched after the discount, some countries discount it, and Belgium has a mixed behaviour)
- The account in which the cash difference resulting of the cash discount should be put.

task- 2983913
related to https://github.com/odoo/odoo/pull/99572

### CODE REVIEW

File: addons/account/models/company.py [hunk 1]
```diff
@@ -671,3 +671,12 @@ def compute_fiscalyear_dates(self, current_date):
 
         return {'date_from': datetime(year=current_date.year, month=1, day=1).date(),
                 'date_to': datetime(year=current_date.year, month=12, day=31).date()}
+
+    def _compute_early_pay_discount_computation(self):
```
Comment (smetl) [inline addons/account/models/company.py:675]: missing @api.depends('country_id')

File: addons/l10n_ae/models/account_chart_template.py [hunk 2]
```diff
@@ -37,3 +37,31 @@ def _load_template(self, company, code_digits=None, account_ref=None, taxes_ref=
                 {'property_tax_payable_account_id': account_ref.get(self.env.ref('l10n_ae.uae_account_202003').id),
                  'property_tax_receivable_account_id': account_ref.get(self.env.ref('l10n_ae.uae_account_100103').id)})
         return account_ref, taxes_ref
+
+    @api.model
+    def _create_cash_discount_loss_account(self, company, code_digits):
+        if not self == self.env.ref('l10n_ae.uae_chart_template_standard'):
```
Comment (smetl) [inline addons/l10n_ae/models/account_chart_template.py:43]: this doesn't work since your method is `api.model`
Instead, add `self.ensure_one()` and check based on the chart template country code

File: addons/l10n_ae/models/account_chart_template.py [hunk 3]
```diff
@@ -37,3 +37,31 @@ def _load_template(self, company, code_digits=None, account_ref=None, taxes_ref=
...
+            return super()._create_cash_discount_loss_account(company, code_digits)
+        cash_discount_loss_account = self.env['account.account'].search([('company_id', '=', company.id), ('code', 'like', '400071')], limit=1)
```
Comment (smetl) [inline addons/l10n_ae/models/account_chart_template.py:45]: This is a bad implementation regarding the chart template logic. It should be fields on the chart template instead

**Review by smetl — CHANGES_REQUESTED** [review]

File: addons/l10n_at/__init__.py [hunk 4]
```diff
@@ -3,3 +3,4 @@
 
 # Copyright (c) 2015 WT-IO-IT GmbH (https://www.wt-io-it.at)
 #                    Mag. Wolfgang Taferner <wolfgang.taferner@wt-io-it.at>
+from . import models
```
Comment (smetl) [inline addons/l10n_at/__init__.py:6]: This line is useless, right?

File: addons/l10n_it/models/account_report.py [hunk 5]
```diff
@@ -1,15 +0,0 @@
...
-from odoo import fields, models
-
-
-class AccountReportExpression(models.AbstractModel):
-    _inherit = "account.report.expression"
-
-    def _get_carryover_target_expression(self, options):
-        if self.report_line_id.code == 'VP14b' and fields.Date.from_string(options['date']['date_to']).month == 12:
-            # For this line, if we are between two years, we want to carry over to vp9 instead of the line set in the XML file (vp8)
-            return self.env.ref('l10n_it.tax_report_line_vp9_applied_carryover')
```
Comment (smetl) [inline addons/l10n_it/models/account_report.py:13]: why did you remove this part?

File: addons/l10n_sg/data/l10n_sg_chart_data.xml [hunk 6]
```diff
@@ -1161,16 +1161,26 @@
...
-        <field name="property_account_income_categ_id" ref="account_account_803" />
+        <field name="property_account_receivable_id" ref="account_account_735"/>
+        <field name="property_account_payable_id" ref="account_account_777"/>
+        <field name="property_account_expense_categ_id" ref="account_account_819"/>
+        <field name="property_account_income_categ_id" ref="account_account_803"/>
         <field name="income_currency_exchange_account_id" ref="account_account_853"/>
         <field name="expense_currency_exchange_account_id" ref="account_account_853"/>
-        <field name="default_pos_receivable_account_id" ref="account_account_737" />
+        <field name="default_pos_receivable_account_id" ref="account_account_737"/>
+        <field name="account_journal_early_pay_discount_loss_account_id" ref="account_account_800"/>
```
Comment (smetl) [inline addons/l10n_sg/data/l10n_sg_chart_data.xml:1183]: Don't you need to update account_account_800 then?

File: addons/l10n_it/models/account_report.py (hunk 5 above)
Comment (casp-odoo) [reply addons/l10n_it/models/account_report.py:13]: just felt like removing it, like a big spring cleaning thing. 

Yeah no, mistake, I'll put it back :)

**Review by smetl — APPROVED** [review]

**Comment by smetl** [comment]
@robodoo pre r+ as asked by the lady
